usecases/schema/handler.go

```diff
@@ -102,38 +112,60 @@ func NewHandler(
 		scaleOut:                scaleoutManager,
 	}
 
-	// TODO-RAFT
-	// m.scaleOut.SetSchemaManager(m)
+	handler.scaleOut.SetSchemaManager(metaReader)
 
-	return m, nil
+	return handler, nil
 }
 
 // GetSchema retrieves a locally cached copy of the schema
-func (m *Handler) GetSchema(principal *models.Principal) (schema.Schema, error) {
-	err := m.Authorizer.Authorize(principal, "list", "schema/*")
+func (h *Handler) GetSchema(principal *models.Principal) (schema.Schema, error) {
+	err := h.Authorizer.Authorize(principal, "list", "schema/*")
 	if err != nil {
 		return schema.Schema{}, err
 	}
 
-	return m.getSchema(), nil
+	return h.getSchema(), nil
 }
 
 // GetSchemaSkipAuth can never be used as a response to a user request as it
 // could leak the schema to an unauthorized user, is intended to be used for
 // non-user triggered processes, such as regular updates / maintenance / etc
-func (m *Handler) GetSchemaSkipAuth() schema.Schema { return m.getSchema() }
+func (h *Handler) GetSchemaSkipAuth() schema.Schema { return h.getSchema() }
 
-func (m *Handler) getSchema() schema.Schema {
-	s := m.metaReader.ReadOnlySchema()
+func (h *Handler) getSchema() schema.Schema {
+	s := h.metaReader.ReadOnlySchema()
 	return schema.Schema{
 		Objects: &s,
 	}
 }
 
-func (m *Handler) Nodes() []string {
-	return m.clusterState.AllNames()
+func (h *Handler) Nodes() []string {
+	return h.clusterState.AllNames()
+}
+
+func (h *Handler) NodeName() string {
+	return h.clusterState.LocalName()
 }
 
-func (m *Handler) NodeName() string {
-	return m.clusterState.LocalName()
+func (h *Handler) UpdateShardStatus(ctx context.Context,
+	principal *models.Principal, class, shard, status string,
+) error {
+	err := h.Authorizer.Authorize(principal, "update",
+		fmt.Sprintf("schema/%s/shards/%s", class, shard))
+	if err != nil {
+		return err
+	}
+
+	return h.metaWriter.UpdateShardStatus(class, shard, status)
+}
+
+func (h *Handler) ShardsStatus(ctx context.Context,
```

Comment:
I'm sure whether we should make 'ShardStatus()' part of the Handler. It doesn't seem to be directly related to schema.